core/src/avm1/globals/shared_object.rs

```diff
@@ -0,0 +1,474 @@
+use crate::avm1::function::{Executable, FunctionObject};
+use crate::avm1::return_value::ReturnValue;
+use crate::avm1::{Avm1, Error, Object, TObject, Value};
+use crate::context::UpdateContext;
+use enumset::EnumSet;
+use gc_arena::MutationContext;
+
+use crate::avm1::shared_object::SharedObject;
+
+use json::JsonValue;
+
+pub fn delete_all<'gc>(
+    _avm: &mut Avm1<'gc>,
+    _action_context: &mut UpdateContext<'_, 'gc, '_>,
+    _this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    log::warn!("SharedObject.deleteAll() not implemented");
+    Ok(Value::Undefined.into())
+}
+
+pub fn get_disk_usage<'gc>(
+    _avm: &mut Avm1<'gc>,
+    _action_context: &mut UpdateContext<'_, 'gc, '_>,
+    _this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<ReturnValue<'gc>, Error> {
+    log::warn!("SharedObject.getDiskUsage() not implemented");
+    Ok(Value::Undefined.into())
+}
+
+/// Serialize an Object and any children to a JSON object
+/// It would be best if this was implemented via serde but due to avm and context it can't
+/// Undefined fields aren't serialized
+fn recursive_serialize<'gc>(
+    avm: &mut Avm1<'gc>,
+    action_context: &mut UpdateContext<'_, 'gc, '_>,
+    obj: Object<'gc>,
+    json_obj: &mut JsonValue,
+) {
+    for k in &obj.get_keys(avm) {
+        let elem = obj.get(k, avm, action_context).unwrap();
```

Comment:
Let's skip the key if it's an `Err` here, or make this function return `Error` and add `?`.